util/chunk/chunk.go

```diff
@@ -59,32 +107,29 @@ func (c *Chunk) MemoryUsage() (sum int64) {
 	return
 }
 
-// addFixedLenColumn adds a fixed length column with elemLen and initial data capacity.
-func (c *Chunk) addFixedLenColumn(elemLen, initCap int) {
-	c.columns = append(c.columns, &column{
+// newFixedLenColumn creates a fixed length column with elemLen and initial data capacity.
+func newFixedLenColumn(elemLen, initCap int) *column {
+	return &column{
 		elemBuf:    make([]byte, elemLen),
 		data:       make([]byte, 0, initCap*elemLen),
 		nullBitmap: make([]byte, 0, initCap>>3),
-	})
+	}
 }
 
-// addVarLenColumn adds a variable length column with initial data capacity.
-func (c *Chunk) addVarLenColumn(initCap int) {
-	c.columns = append(c.columns, &column{
+// newVarLenColumn creates a variable length column with initial data capacity.
+func newVarLenColumn(initCap int, old *column) *column {
+	estimatedElemLen := 4
+	// For varLenColumn(e.g. varchar) we could not take the accuracy length of element,
+	// so, in first executor.Next we using a experience value --- 4(so it maybe make `runtime.growslice`)
+	// but in continue Next call we estimated length as AVG x 1.5 elemLen of previous call.
+	if old != nil && old.length != 0 {
+		estimatedElemLen = (len(old.data) + len(old.data)/2) / old.length
```

Comment:
why the estimated length for the var-length element is changed?

Reply:
To reduce the probability of growing slice when appending string.

Reply:
`len(old.data)/old.length` is enough to estimated the average length of the strings to be appended in the new `column`, no need to moltiply `1.5`.

Reply:
@coocood @zz-jason PTAL... I have conitune PR wait this...